src/cls/rgw/cls_rgw_ops.h

```diff
@@ -1002,56 +1002,86 @@ struct cls_rgw_lc_get_next_entry_op {
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_op)
 
 struct cls_rgw_lc_get_next_entry_ret {
-  pair<string, int> entry;
+private:
+  mutable pair<string, int> compat_entry; // only for backward compatibility
+public:
+  cls_rgw_lc_entry entry;
 
   cls_rgw_lc_get_next_entry_ret() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
+    compat_entry = make_pair(entry.bucket, entry.status);
+    ::encode(compat_entry, bl); 
     ::encode(entry, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(bufferlist::iterator& bl) {
-    DECODE_START(1, bl);
-    ::decode(entry, bl);
+    DECODE_START(2, bl);
+    ::decode(compat_entry, bl);
+    entry.bucket = compat_entry.first;
```

Comment:
Here, if version < 2, decode the compat entry, otherwise decode the real entry.  Don't do both.  This goes for all the decodes.

Reply:
as @yehudasa previously commented,  we should keep backward compatible structure encoding. If we do not encode the compat entry,  during upgrading, if we upgrading osd firstly, then the old version rgw cannot decode the entry correctly.